credoai/evaluators/utils/validation.py

```diff
@@ -50,6 +59,48 @@ def check_existence(obj, name=None):
         raise ValidationError(message)
 
 
+def check_nulls_by_data_type(data):
+    nulls = False
+    if isinstance(data, (pd.DataFrame, pd.Series)):
+        nulls = data.isnull().to_numpy().any()
+    if isinstance(data, np.ndarray):
+        nulls = np.isnan(data).any()
+    if isinstance(data, tf.Tensor):
+        nulls = tf.reduce_any(tf.math.is_nan(data))
+    if isinstance(
+        data, (tf.data.Dataset, tf.keras.utils.Sequence)
+    ) or inspect.isgeneratorfunction(data):
+        message = """
+        Evaluator Validation: Checking for nulls in generator-based or mapped data is not currently
+        supported. Please be sure to sanitize your data. Downstream errors may arise due to nulls in 
+        image or other tensor data.
+        """
+        global_logger.warning(message)
+    return nulls
+
+
+#################################
+# Checking evaluator requirements
+#################################
+
+
+def check_data_for_nulls(obj, name, check_X=True, check_y=True, check_sens=True):
```

Comment:
Not necessarily. This could refer to a Tensor or other object that doesn't have a `name` attribute. The `name` variable is only used for error outputting to say "This thing you provided (which you called `name`) contains nulls when it shouldn't"